### PR TITLE
[C++ for OpenCL] Changed the structure or the document.

### PR DESCRIPTION
**Pull request by AnastasiaStulova**
Make restriction section to describe general differences to C++17.

### CODE REVIEW

File: cxx4opencl/diff2cxx.txt [hunk 1]
```diff
@@ -2,11 +2,10 @@
...
 // http://creativecommons.org/licenses/by/4.0/
 
-[[restrictions]]
-=== Restrictions
+[[diff2cxx]]
+=== Difference to {cpp}
 
-This section describes various restrictions and limitations for
-language features from OpenCL v2.0 and {cpp}17.
+{cpp} for OpenCL supports the majority of standard {cpp}17 features. However, there are some differences that are documented in this section.
```
Comment (neildhickey) [inline cxx4opencl/diff2cxx.txt:8]: should be , however.

Comment (AnastasiaStulova) [reply cxx4opencl/diff2cxx.txt]: updated!

Also added a bit more explanation about the next section.

File: cxx4opencl/intro.txt [hunk 2]
```diff
@@ -7,13 +7,22 @@
 
 This language is built on top of OpenCL C v2.0 and {cpp}17 enabling
 most of regular {cpp} features in OpenCL kernel code. Most functionality
-from OpenCL C and {cpp} is inherited. Restrictions are documented
-in <<restrictions, _Restrictions_>> sections.
+from OpenCL C and {cpp} is inherited. The following two sections of this
```
Comment (svenvh) [inline cxx4opencl/intro.txt:10]: ```This document includes a description of <<diff2openclc, _the differences to OpenCL C_>> and <<diff2cxx, _the differences to {cpp}_>>.```

Please also make sure the lines are wrapped at 80 columns.

File: cxx4opencl/intro.txt [hunk 3]
```diff
@@ -7,13 +7,22 @@
...
-documented in the OpenCL C and {cpp}17 specifications. This is mainly related
-to interactions between OpenCL and {cpp} language features.
+The majority of content covers the behavior that is not documented in
+the OpenCL C v2.0 section 6 and {cpp}17 specifications. This is mainly
+related to interactions between OpenCL and {cpp} language features.
 
 == The {cpp} for OpenCL Programming Language 
+
+This programming language inherits features from `OpenCL C v2.0 s6` as well
+as C++17 (`ISO/IEC 14882:2017`). Detailed aspects of OpenCL and C++ are not
```
Comment (svenvh) [inline cxx4opencl/intro.txt:20]: s/C++/{cpp}/g

File: cxx4opencl/intro.txt [hunk 4]
```diff
@@ -7,13 +7,22 @@
...
+described in this document as they can be found in their official
+specifications.
  
-This chapter documents various language features of {cpp} for OpenCL.
+This section documents various language features of {cpp} for OpenCL that are
+not covered in either OpenCL or C++ specifications, in particular:
```
Comment (svenvh) [inline cxx4opencl/intro.txt:25]: {cpp}

File: cxx4opencl/intro.txt (hunk 2 above)
Comment (AnastasiaStulova) [reply cxx4opencl/intro.txt:10]: Hmm but it is only 2 sections that explain the difference not the entire document. What is it that you want to be changed here exactly?

Comment (svenvh) [reply cxx4opencl/intro.txt:10]: I find "the following two sections" a bit ambiguous, as the reader may also think that it refers to the next two paragraphs.

Comment (AnastasiaStulova) [reply cxx4opencl/intro.txt]: Ok, I provided more description now.

File: cxx4opencl/intro.txt [hunk 5]
```diff
@@ -7,13 +7,32 @@
...
-in <<restrictions, _Restrictions_>> sections.
+from OpenCL C and {cpp} is inherited.
 
-The main aim of this document is to describe behavior that is not
-documented in the OpenCL C and {cpp}17 specifications. This is mainly related
-to interactions between OpenCL and {cpp} language features.
+This document describes the programming language in details. It is not
+structured as a standalone document, but rather an addition to `OpenCL C
+v2.0 s6` and {cpp}17 (`ISO/IEC 14882:2017`). Where necessary this document
+refers to the specifications of those languages accordingly. For the full
```
Comment (svenvh) [inline cxx4opencl/intro.txt:15]:
```suggestion
refers to the specifications of those languages accordingly. For a full
```

File: cxx4opencl/intro.txt [hunk 6]
```diff
@@ -7,13 +7,32 @@
...
+understanding of {cpp} for OpenCL it is required to get familiar with the
+specifications or other documentation of both languages that {cpp} for OpenCL
+is build upon.
```
Comment (svenvh) [inline cxx4opencl/intro.txt:18]:
```suggestion
is built upon.
```

File: cxx4opencl/intro.txt [hunk 7]
```diff
@@ -7,13 +7,32 @@
...
+from OpenCL C and {cpp} is inherited.
 
-The main aim of this document is to describe behavior that is not
-documented in the OpenCL C and {cpp}17 specifications. This is mainly related
-to interactions between OpenCL and {cpp} language features.
+This document describes the programming language in details. It is not
+structured as a standalone document, but rather an addition to `OpenCL C
+v2.0 s6` and {cpp}17 (`ISO/IEC 14882:2017`). Where necessary this document
+refers to the specifications of those languages accordingly. For the full
+understanding of {cpp} for OpenCL it is required to get familiar with the
```
Comment (svenvh) [inline cxx4opencl/intro.txt:16]: ```A full understanding of {cpp} for OpenCL requires familiarity with the```

**Pull request merged**